### DISCUSSION TITLE
How to set ModelAdmin types

### DISCUSSION THREAD

**Discussion by knyghty** [sbdchd/django-types#161] [Q&A] [answered]
```python
@admin.register(Foo)
class FooAdmin(admin.ModelAdmin):
    ...
```

Pyright:
```
error: Expected type arguments for generic class "ModelAdmin" (reportMissingTypeArgument)
```

With:

```python
@admin.register(Foo)
class FooAdmin(admin.ModelAdmin[Foo]):
    ...
```

Pyright is happy but now the code doesn't run because the real `ModelAdmin` class is not subscriptable. I'm not sure what I need to do to get both to be happy. Possibly something in `if TYPE_CHECKING:` but I can't figure out quite the right incantation.

Any ideas here?

**Comment by sbdchd** [comment, marked as answer]
Yeah if it works at compile time but explodes at runtime you can monkey patch the underlying class to have a `__class_getitem__` method:

```python
for cls in [admin.ModelAdmin]:
    cls.__class_getitem__ = classmethod(lambda cls, *args, **kwargs: cls)  # type: ignore [attr-defined]
```
